### PR TITLE
Rename and tweak deploy scripts to enable CI deploys.

### PR DESCRIPTION
**Pull request by adewinter**
re-organized scripts to so that the following deploy groups have deploy/rollback scripts:
* canary_tier_1
* canary_tier_2
* production

Scripts now also have a `--force` or `-f` option that can be used
to trigger "non-interactive" mode, ideal for use with CI builds.

### CODE REVIEW

**Comment by adewinter** [comment]
Before I can merge this I need to make some discussed changes:

* Only upload minified versions
* Ensure these files include a reference to what version they are (either in filename or as a comment at top of file)
* Make rollback go back to the prior version given the above.

**Comment by adewinter** [comment]
@pevner-p2 @Detect The current scripts let you roll back to the immediately prior version (see e.g. [rollback_canary_tier_2.sh](https://github.com/ProdPerfect/prodperfect-keen-tracking.js/pull/40/files#diff-ef21d898bffe92655503264417c9484e)) but anything beyond that will require us to invest a lot more time:

* Something needs to track all the past versions that have been deployed to the bucket
* Needs to give the deployer the ability to roll back to any one of those past versions
* etc

I think what we have in this PR (making an immediate backup of existing deployed artifact, uploading a versioned filename of the new artifact) should be enough to get us by for the forseeable future.

What do you think?

File: build_scripts/deploy_production.sh [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
-  echo "copying current min file to last";
-  aws s3 cp "s3://prodperfect-keen-js/${MIN_FILE}" "s3://prodperfect-keen-js/keen-tracking_last.min.js" --region us-east-2 --acl public-read
+  echo "uploading a versioned build of ${FILE_NAME}.min to s3://${S3_BUCKET}/${VERSIONED_FILE}";
+  aws s3 cp "./dist/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${VERSIONED_FILE}" --region us-east-2 --acl public-read
+ 
+  echo "Making backup of existing '${MINIFIED_FILE}' file.";
+  aws s3 cp "s3://${S3_BUCKET}/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${BACKUP_FILE}" --region us-east-1 --acl public-read
+
+  echo "Uploading latest build of '${MINIFIED_FILE}'."  
+  aws s3 cp "./dist/${MINIFIED_FILE}" "s3://${S3_BUCKET}/${MINIFIED_FILE}" --region us-east-1 --acl public-read
```
Comment (Detect) [inline build_scripts/deploy_production.sh]: some files are put on us-east-1, another is on us-east-2. is this correct? thoughts on making an aws region variable at the top?

File: build_scripts/rollback_canary_tier_1.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          [Nn]* ) echo "Ok, exiting."; exit;;
+          * ) echo "Please answer yes or no.";;
+      esac
+  done
+fi
+
+if aws --version &> /dev/null
+then
+  echo "rolling back canary";
+  aws s3 cp "s3://${S3_BUCKET}/keen-tracking_last.min.js" "s3://${S3_BUCKET}/${MIN_FILE}" --region us-east-2 --acl public-read
```
Comment (Detect) [inline build_scripts/rollback_canary_tier_1.sh]: note: in the other file, variable `BACKUP_FILE` was used and in this file it's hardcoded.

File: build_scripts/deploy_production.sh (hunk 1 above)
Comment (adewinter) [reply build_scripts/deploy_production.sh]: Yeah I wasn't sure what we wanted here, I just followed on from what was there previously.  @pevner-p2 do you have feelings about what region this should be?

Comment (pevner-p2) [reply build_scripts/deploy_production.sh]: I'd prefer, for my sanity, that everything were inside us-east-1, if possible.

**Pull request merged**